src/sgl/device/device.cpp

```
@@ -988,7 +1020,12 @@ uint64_t Device::submit_command_buffers(
         }
     }
 
-    return m_global_fence->signaled_value();
+    const uint64_t submit_id = m_global_fence->signaled_value();
+
+    for (CommandBuffer* command_buffer : command_buffers)
+        command_buffer->_notify_submitted(submit_id);
+
+    return submit_id;
```

Cppcheck (2.20.0)

[style] The function 'name_override' is never used.

(unusedFunction)